lms/views/error.py

```diff
@@ -1,66 +1,63 @@
 
-    @view_config(context=Exception)
-    def error(self):
-        # In debug mode re-raise exceptions so that they get printed in the
-        # terminal.
-        if asbool(self.request.registry.settings.get('debug')):
-            raise self.exc
+        If code raises an HTTP client or server error we assume this was
+        deliberately raised We show the user an error page including specific
+        error message but _do not_ report the error to Sentry
+        """
+        self.request.response.status_int = self.exc.status_int
+        return {"message": str(self.exc)}
 
-        self.request.response.status_int = 500
+    @exception_view_config(LTILaunchError)
+    def missing_lti_param_error(self):
+        """
+        Handle LTILaunchErrors.
 
-        # If code raises a non-HTTPException exception we assume it was a bug
-        # and:
-        # 1. Show the user a generic error page
-        # 2. Report the details of the error to Sentry
+        If code raises an LTILaunchError we assume this was deliberately raised
+        because an invalid LTI launch request was received. For example a
+        required LTI launch parameter was missing. We return a 400 Bad Request
+        response, show the user an error page with the specific error message,
+        and report the issue to Sentry.
+        """
+        self.request.response.status_int = 400
         self.request.raven.captureException()
-        return {'message': _("Sorry, but something went wrong. "
-                             "The issue has been reported and we'll try to "
-                             "fix it.")}
+        return {"message": str(self.exc)}
 
-
-    @view_config(context=LTILaunchError)
-    def missing_lti_param_error(self):
+    @exception_view_config(Exception)
+    def error(self):
         """
-        Catch MissingLTILaunchParamError and MissingLTIContentItemParamError, render a 400 page and report the exception to Sentry.
-
-        If code raises a MissingLTILaunchParamError it means that a required parameter was missing
-        from an LTI launch request that we received.
-        If code raises a MissingLTIContentItemParamError it means that a required parameter was missing
-        from an LTI content item selection request that we received:
+        Handle an unexpected exception.
 
-        1. Show the user an error page including specific error message
-        2. Report the error to Sentry
+        If the code raises an unexpected exception (anything not caught by any
+        of the more specific exception views above) then we assume it was a
+        bug.  We show the user a generic error page and report the exception to
+        Sentry.
         """
-        self.request.response.status_int = 400
+        self.request.response.status_int = 500
         self.request.raven.captureException()
-        return {'message': str(self.exc)}
+        return {
+            "message": _(
+                "Sorry, but something went wrong. "
```

Comment:
I think we should try to make this consistent with our other error message from h: 
 > Uh-oh, something went wrong!
 > We're very sorry, our application wasn't able to load this page. The team has been notified and we'll fix it shortly.
 > If the problem persists or you'd like more information please contact support@hypothes.is.

Reply:
I think this needs to be fixed in a separate PR. The problem is that the h error page has different templating and styling than the LMS one. It uses a separate `<h1>` and `<p>` for the above error message. The LMS app would wrap it all together in one paragraph and center-align it all. Needs a follow-up PR to improve the design of the error page